src/details/ArborX_DetailsDistributedTreeImpl.hpp

```diff
@@ -32,8 +32,30 @@
 namespace ArborX
 {
 
+struct PairIndexRank
```

Comment:
I am not so keen on making it a "public" type.  I would define it in `Details::`

Reply:
That is the return type of the default distributed query, and is user visible. I believe it belongs to the proper `ArborX` namespace.